flexmeasures/data/models/planning/utils.py

```diff
@@ -157,18 +157,30 @@ def fallback_charging_policy(
         < 0
     ):
         # start discharging to get as close as possible to the next target
-        return discharge_schedule
+        return idle_after_reaching_target(
+            discharge_schedule, device_constraints["equals"]
+        )
     if (
         device_constraints["max"].first_valid_index() is not None
         and device_constraints["max"][device_constraints["max"].first_valid_index()] < 0
     ):
         # start discharging to try and bring back the soc below the next max constraint
-        return discharge_schedule
+        return idle_after_reaching_target(discharge_schedule, device_constraints["max"])
     if (
         device_constraints["min"].first_valid_index() is not None
         and device_constraints["min"][device_constraints["min"].first_valid_index()] > 0
     ):
         # start charging to try and bring back the soc above the next min constraint
-        return charge_schedule
+        return idle_after_reaching_target(charge_schedule, device_constraints["min"])
     # stand idle
     return idle_schedule
+
+
+def idle_after_reaching_target(schedule: pd.Series, target: pd.Series) -> pd.Series:
+    """Stop planned (dis)charging after target is reached (or constraint is met)."""
+    first_target = target[target.first_valid_index()]
+    if first_target > 0:
```

Comment:
Isn't it a more important distinction here to decide if we needed to add (`first_target > some_initial_state`) versus needing to subtract (`first_target < some_initial_state`)?

Reply:
Correct, but the initial state is 0 by definition. The problem formulation is translated to ensure this.

Reply:
Maybe worth a small comment